From 43bda2761f6efb41d565530f9ff75c3b0fe3a568 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Date: Fri, 7 May 2021 23:54:23 -0700
Subject: [PATCH 1525/1921] drivers: soc: marvell: Add config option for serdes
 diagnostics

Add new config option to build serdes diagnostics as builtin
or as module.

Signed-off-by: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Change-Id: Ib3c3a30a21d97b5671b53113fa1fa92a597e01b0
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/51349
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/Kconfig                   | 9 +++++++++
 drivers/soc/marvell/Makefile                  | 2 +-
 drivers/soc/marvell/octeontx2-serdes/Makefile | 2 +-
 3 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/Kconfig b/drivers/soc/marvell/Kconfig
index 2df85dbaeee6..9ef970ac3083 100644
--- a/drivers/soc/marvell/Kconfig
+++ b/drivers/soc/marvell/Kconfig
@@ -135,4 +135,13 @@ config HW_CSR_ACCESS
 	help
 	  provides an interface which enables user space utility to read from or to write
 	  to HW Device Config Space Registers.
+
+config OCTEONTX_SERDES
+	tristate "OcteonTX2 serdes debug commands"
+	depends on ARM64
+	help
+	  Provides debugfs commands which allows to collect serdes information,
+	  capture eye, use PRBS, configure serdes in loopback mode, and
+	  read/write TX tuning parameters.
+
 endmenu
diff --git a/drivers/soc/marvell/Makefile b/drivers/soc/marvell/Makefile
index b671af8df91d..60e8c7cdffe9 100644
--- a/drivers/soc/marvell/Makefile
+++ b/drivers/soc/marvell/Makefile
@@ -6,6 +6,6 @@ obj-y		+= octeontx2-sdp/
 obj-y		+= octeontx2-npa/
 obj-y		+= octeontx2-llc/
 obj-y 		+= csr-access/
-obj-y          += octeontx2-serdes/
+obj-y 		+= octeontx2-serdes/
 obj-$(CONFIG_OCTEONTX_INFO) += octeontx_info.o
 obj-$(CONFIG_MVMDIO_UIO) += mvmdio_uio.o
diff --git a/drivers/soc/marvell/octeontx2-serdes/Makefile b/drivers/soc/marvell/octeontx2-serdes/Makefile
index f521770fe5b1..d7606038b597 100644
--- a/drivers/soc/marvell/octeontx2-serdes/Makefile
+++ b/drivers/soc/marvell/octeontx2-serdes/Makefile
@@ -3,5 +3,5 @@
 # Makefile for Marvell's OcteonTX2 serdes diagnostic commands
 #
 
-obj-y += serdes_debugfs.o
+obj-$(CONFIG_OCTEONTX_SERDES) += serdes_debugfs.o
 
-- 
2.31.1

